### PR TITLE
Added Tribler request dispatcher with a multiple priority request pools

### PR DESCRIPTION
**Pull request by xoriole**
Now request dispatcher dispatches the request to available request worker (QNetworkAccessManager) to process the request.

### CODE REVIEW

File: TriblerGUI/tribler_request_manager.py [hunk 1]
```diff
@@ -37,104 +37,123 @@ def __init__(self, max_outstanding=50, timeout=15):
...
-                if self.queue[i] == insert_find:
-                    insert_point = i + (0 if insert_step == 1 else 1)
-                    break
-        # Update pointers
+        queue_item = (request_manager, endpoint, read_callback, data, method, capture_errors, time())
+        if priority == QueuePriorityEnum.CRITICAL:
+            if len(self.critical_queue) < self.max_outstanding:
+                self.critical_queue.append(queue_item)
+            else:
+                # no space left, reduce the priority to high
```
Comment (qstokkink) [inline TriblerGUI/tribler_request_manager.py]: Why not just boot the lowest priority request and add this to the queue as CRITICAL priority?

**Comment by xoriole** [comment]
retest this please

File: TriblerGUI/tribler_request_manager.py (hunk 1 above)
Comment (xoriole) [reply TriblerGUI/tribler_request_manager.py]: Yes, now lowest priority request is booted out if no space is left in other queues.

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Pull request merged**